src/main/scala/rocket/ICache.scala

```diff
@@ -400,6 +400,7 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
       }
   }
 
+  assert (!(io.s2_kill && tl_out.a.ready && tl_out.a.valid))
```

Comment:
I don't think this is right.  tl_out.a.ready & valid can legitimately happen when s2_kill is high, for the purposes of sending a prefetch hint.